### PR TITLE
Test/e2e

### PR DESCRIPTION
**Pull request by joshLong145**
Changes

**Architecture**

```
Before                          After
┌──────────┐                    ┌──────────┐    ┌──────────┐
│ main.rs  │ (mod app, ...)     │ main.rs  │───▶│  lib.rs  │ (pub mod app, ...)
│ duped fns│                    │ bin-only │    │ shared   │
└──────────┘                    └──────────┘    └──────────┘
                                                     ▲
                                                ┌────┴─────┐
                                                │ tests/   │
                                                │  e2e.rs  │
                                                │  helpers/ │
                                                └──────────┘
```

- Created `src/lib.rs` — enables integration tests to import project types
- Moved shared modules (`app`, `coverage`, `ingest`, `parser`, `symbols`, `tracking`) into the lib crate
- Kept binary-only modules (`events`, `ui`, `serena`, `skill`) in `main.rs`
- Deleted 4 duplicated functions from `main.rs`; made the `app.rs` versions `pub`

**Test helpers** (`tests/helpers/mod.rs`)

Shared builders (`sym`, `file`, `project`, `tool_call`, `jsonl_assistant`, etc.) used by unit tests via `#[path]` and by integration tests directly. Keeps test setup DRY across all modules.

**New tests by module**

| Module | New | What's covered |
|--------|-----|----------------|
| `tracking` | 6 | record upgrade/downgrade, stale transitions, defaults |
| `coverage` | 8 | percentages, counting, report sorting, formatter output |
| `app` | 12 | free functions + `process_agent_event`, `rebuild_tree_rows` |
| `ingest/claude` | 17 | all 15 `map_tool_call` arms + edge cases |
| `parser/rust` | 4 | function, struct+impl, nested module, empty file |
| `tests/e2e` | 8 | full pipeline, targeting, depth upgrades, stale detection |

**CI** (`.github/workflows/ci.yml`)

Runs `cargo build` + `cargo test` on every PR to `main` and every push to `main`. Uses `rust-cache` for fast builds

### CODE REVIEW

File: src/ingest/claude.rs [hunk 1]
```diff
@@ -582,6 +582,11 @@ impl LogTailer {
     }
 }
 
+#[cfg(test)]
+#[path = "../../tests/helpers/mod.rs"]
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (joshLong145) [reply src/ingest/claude.rs]: This was introduced to bridge between the `tests/` directory and the actual source implementations. with the `#[cfg(test)]` directive we only use this module when configured to run tests. so static analyzers will think of this as an unused import.
